logger/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+use constants::DRIVER_NAME;
+use cstr::to_widechar_ptr;
+use log::LevelFilter;
+use log4rs::encode::pattern::PatternEncoder;
+use log4rs::Handle;
+use log4rs::{
+    append::rolling_file::{
+        policy::compound::{
+            roll::fixed_window::FixedWindowRoller, trigger::size::SizeTrigger, CompoundPolicy,
+        },
+        RollingFileAppender,
+    },
+    config::{Appender, Config, Root},
+};
+use shared_sql_utils::odbcinst::SQLGetPrivateProfileStringW;
+use std::path::{Path, PathBuf};
+
+const LOG_FILE_SIZE: u64 = 1024 * 500;
+
+#[derive(Debug)]
+pub struct Logger {
+    handle: Handle,
+    log_dir: PathBuf,
+}
+
+impl Logger {
+    pub fn new() -> Option<Self> {
+        // Due to numerous reasons why the logger could fail to initialize, we wrap it in a catch_unwind
+        // so that logger failure does not cause our dll to crash.
+        match std::panic::catch_unwind(|| {
+            let mut buffer = [0u16; 1024];
+            unsafe {
+                SQLGetPrivateProfileStringW(
```

Comment:
I wonder about our level of abstraction here. Instead of calling odbc functions here in the `logger` crate, `Logger::new()` can take `driver_path` as an argument, and we can have a util function in the odbc crate that does this work and calls `Logger::new()` with the appropriate `driver_path` value.

Reply:
I thought about this. My worry was this would add weight to the initial `SQLAllocHandle` function, and my design hope was to initialize the logger such that it was completely segmented from the main driver and avoid interfacing with the SQL Installer functions as much as possible (supporting DSN connections being an area where it absolutely needs to).

That said, I was looking for a reason to change the entry to accept a path to make testing it easier, and this is reason enough!